addon/vector.js

```diff
@@ -74,7 +74,39 @@ class CorpusCollector extends PageVisitor {
         }
         if (vector !== undefined) {
             this._vectors.push(vector);
-            this.setCurrentStatus({message: 'vectorized', isFinal: true});
+
+            // Check if any of the rules didn't run or returned null.
+            // This presents as an undefined value in a feature vector.
+            const nullFeatures = await this.nullFeatures(vector.nodes);
+            if (nullFeatures) {
+                console.log(nullFeatures);
+                this.setCurrentStatus({
+                    message: 'failed: rule(s) ' + nullFeatures + ' returned null values',
```

Comment:
I'm also a little confused by the name choice here of `nullFeatures`, because when I think of `feature`, I think of things like `"overlay"`, `"image"`, `"price"`, `"title"`. But doesn't `nullFeatures` here represent a list of rule names?

Reply:
I'd be interested to see what errors eslint throws if we add in `prefer-template`. One nice thing about template literals is that it dodges the quirks JS's `+` operator has when used with different types. I haven't spent long enough with template literals to state a hard-and-fast preference, however.

Reply:
I do like template literals quite a bit, as I also like f-strings in Python.

Reply:
I believe `"overlay"`, `"image"`, `"price"`, and `"title"` are all `type`s in Fathom lingo. Here feature is used in the [machine learning context.](https://en.wikipedia.org/wiki/Feature_(machine_learning)), and does, indeed, represent the output of our rules. Each rule output is a feature of an evaluated element, and the feature vectors are what we feed into our neural network.